superset-frontend/src/views/CRUD/chart/ChartList.tsx

```diff
@@ -164,7 +164,8 @@ function ChartList(props: ChartListProps) {
   const [preparingExport, setPreparingExport] = useState<boolean>(false);
 
   const { userId } = props.user;
-  const userKey = getFromLocalStorage(userId?.toString(), null);
+  // TODO: Fix usage of localStorage keying on the user id
+  const userKey = dangerouslyGetItemDoNotUse(userId?.toString(), null);
```

Comment:
I think `userKey` may actually be `userProfile` or `userSettings`. We can also do type coercing to avoid calling the private dangerous functions:

```ts
const userProfile = getItem(String(userId || '') as "FIXME_user_profile_by_user_id", null);

type LocalStorageValues {
  // it's a bad practice to use the raw user id as storage keys
  // we should at least prefix them.
  FIXME_user_profile_by_user_id: {
    thumbnails: boolean;
  }
}
```

Reply:
i'll update the variable name. personally i prefer using the dangerous function here though, since it seems a bit easier to parse than the type assertion (that's clearly not true)